common/src/main/scala/org/neo4j/spark/writer/DataWriterMetrics.scala

```diff
@@ -0,0 +1,140 @@
+package org.neo4j.spark.writer
+
+import org.apache.spark.sql.connector.metric.{CustomMetric, CustomSumMetric, CustomTaskMetric}
+import org.neo4j.driver.summary.SummaryCounters
+import org.neo4j.spark.writer.DataWriterMetrics.{LABELS_ADDED, LABELS_REMOVED, NODES_CREATED, NODES_DELETED, PROPERTIES_SET, RECORDS_WRITTEN, RELATIONSHIPS_CREATED, RELATIONSHIPS_DELETED}
+
+import java.util.concurrent.atomic.AtomicLong
+
+case class DataWriterMetric(name: String, value: Long) extends CustomTaskMetric {
+}
+
+class DataWriterMetrics private(
+                                 recordsProcessed: AtomicLong,
+                                 nodesCreated: AtomicLong,
+                                 nodesDeleted: AtomicLong,
+                                 relationshipsCreated: AtomicLong,
+                                 relationshipsDeleted: AtomicLong,
+                                 propertiesSet: AtomicLong,
+                                 labelsAdded: AtomicLong,
+                                 labelsRemoved: AtomicLong) {
+
+  def applyCounters(recordsWritten: Long, counters: SummaryCounters): Unit = {
+    this.recordsProcessed.addAndGet(recordsWritten)
+    this.nodesCreated.addAndGet(counters.nodesCreated())
+    this.nodesDeleted.addAndGet(counters.nodesDeleted())
+    this.relationshipsCreated.addAndGet(counters.relationshipsCreated())
+    this.relationshipsDeleted.addAndGet(counters.relationshipsDeleted())
+    this.propertiesSet.addAndGet(counters.propertiesSet())
+    this.labelsAdded.addAndGet(counters.labelsAdded())
+    this.labelsRemoved.addAndGet(counters.labelsRemoved())
+  }
+
+  def metricValues(): Array[CustomTaskMetric] = {
+    List[CustomTaskMetric](
+      DataWriterMetric(RECORDS_WRITTEN, recordsProcessed.longValue()),
+      DataWriterMetric(NODES_CREATED, nodesCreated.longValue()),
+      DataWriterMetric(NODES_DELETED, nodesDeleted.longValue()),
+      DataWriterMetric(RELATIONSHIPS_CREATED, relationshipsCreated.longValue()),
+      DataWriterMetric(RELATIONSHIPS_DELETED, relationshipsDeleted.longValue()),
+      DataWriterMetric(PROPERTIES_SET, propertiesSet.longValue()),
+      DataWriterMetric(LABELS_ADDED, labelsAdded.longValue()),
+      DataWriterMetric(LABELS_REMOVED, labelsRemoved.longValue())
+    ).toArray
+  }
+
+}
+
+object DataWriterMetrics {
+  final val RECORDS_WRITTEN = "recordsWritten"
```

Comment:
I don't know that are the naming conventions spark have for metrics, but looking at existing (for instance) https://spark.apache.org/docs/latest/monitoring.html#executor-metrics, spark sometimes combines them by groups like `memoryMetrics.*` or `peakMemoryMetrics`. I wonder if it makes sense to prefix our custom metrics with something like `neo4jMetrics.*`?

Reply:
I like this idea. The only potential gotcha is that the name of custom metrics that seem to surface are based on the descriptions we provide, not the name.
WDYT @ali-ince?

Reply:
Yeah, I agree that it's a nice idea. Let's use namespaced naming scheme as @venikkin suggests, and we can update the descriptions in the future if we need to.